linkatos/activities.py

```diff
@@ -11,34 +11,40 @@ def is_empty(events):
 def is_url(url_cache):
     return url_cache is not None
 
-def event_consumer(expecting_url, url_cache, slack_client,
-                   fb_credentials, firebase):
 
+def remove_url_from(url_cache_list, index):
+    url_cache_list.pop(index)
+
+
+def is_not_from_bot(bot_id, user_id):
+    return not bot_id == user_id
+
+
+def event_consumer(url_cache_list, slack_client, bot_id,
+                   fb_credentials, firebase):
     # Read slack events
     events = slack_client.rtm_read()
 
     if is_empty(events):
-        return (expecting_url, url_cache)
+        return url_cache_list
 
     for event in events:
-        print(event)
-        print('expecting_url: ', expecting_url)
-
-        if expecting_url and event['type'] == 'message':
-            url_cache = parser.parse_url_message(event)
+        if event['type'] == 'message':
+            new_url_cache = parser.parse_url_message(event)
 
-            if url_cache is not None:
-                printer.ask_confirmation(url_cache, slack_client)
-                expecting_url = False
+            if is_url(new_url_cache) and is_not_from_bot(bot_id,
+                                                         new_url_cache['user']):
+                url_cache_list.append(new_url_cache)
+                printer.ask_confirmation(new_url_cache, slack_client)
 
-        if not expecting_url and event['type'] == 'reaction_added':
+        if event['type'] == 'reaction_added' and len(url_cache_list) > 0:
             reaction = parser.parse_reaction_added(event)
+            index = react.is_confirmation(reaction['reaction'], url_cache_list,
+                                          reaction['to_id'])
 
-            if react.is_confirmation(reaction['reaction'],
-                                     url_cache['id'],
-                                     reaction['to_id']):
-                react.handle(reaction['reaction'], url_cache['url'],
+            if react.is_expected_reaction(index):
+                react.handle(reaction['reaction'], url_cache_list[index]['url'],
                              fb_credentials, firebase)
-                expecting_url = True
+                remove_url_from(url_cache_list, index)
```

Comment:
At some point we need to deal with the possibility `react.handle` rises an error.